Build is broken for main due to conflicting PRs

https://github.com/dotnet/runtime/pull/102636 and https://github.com/dotnet/runtime/pull/102034 got merged in parallel and conflict. The latter made GenericCache generic and the former uses the non-generic.
```
D:\repos\runtime\src\coreclr\System.Private.CoreLib\src\System\RuntimeType.CoreCLR.cs(3970,17): error CS0305: Using the generic type 'GenericCache<TKey, TValue>' requires 2 type arguments [D:\repos\runtime\src\coreclr\System.Private.CoreLib\System.Private.CoreLib.cspro
j]
D:\repos\runtime\src\coreclr\System.Private.CoreLib\src\System\RuntimeType.CoreCLR.cs(3972,29): error CS0122: 'RuntimeType.ActivatorCache.ActivatorCache(RuntimeType)' is inaccessible due to its protection level [D:\repos\runtime\src\coreclr\System.Private.CoreLib\Syste
m.Private.CoreLib.csproj]
D:\repos\runtime\src\coreclr\System.Private.CoreLib\src\System\RuntimeType.CoreCLR.cs(3973,17): error CS0305: Using the generic type 'GenericCache<TKey, TValue>' requires 2 type arguments [D:\repos\runtime\src\coreclr\System.Private.CoreLib\System.Private.CoreLib.cspro
j]
```
cc: @jkoritzinsky, @AaronRobinsonMSFT

--- a/src/coreclr/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs
+++ b/src/coreclr/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs
@@ -3947,7 +3947,7 @@ namespace System
             }
 
             // We reuse ActivatorCache here to ensure that we aren't always creating two entries in the cache.
-            object? obj = GetOrCreateCacheEntry<ActivatorCache>().CreateUninitializedObject(this);
+            object? obj = cache.CreateUninitializedObject(this);
             try
             {
                 cache.CallRefConstructor(obj);
@@ -3967,11 +3967,7 @@ namespace System
         {
             Debug.Assert(IsValueType);
 
-            if (GenericCache is not ActivatorCache cache)
-            {
-                cache = new ActivatorCache(this);
-                GenericCache = cache;
-            }
+            ActivatorCache cache = GetOrCreateCacheEntry<ActivatorCache>();
 
             if (!cache.CtorIsPublic)
             {

